### ISSUE TITLE
[Collection] missing validations

### ISSUE THREAD

**Issue by mkuhring** [bihealth/altamisa#17]
Collecting of validations we could and mostly should add at some point.

### Things to prevent
- [ ] Data Files can't have Characteristics (separate them from Material?)
- [ ] Empty Contacts/Publications columns should result in no object (currently empty object)

### Thinks to check
Following validations will require a proper association of the study/assay infos from the investigation file and the corresponding study sheet.
- [ ] Is a factor type defined in the investigation (section STUDY FACTORS)?
- [ ] Is a parameter type defined in the investigation (section STUDY PROTOCOLS)?
- [ ] Are all expected sections available for Investigation/Study?
- [ ] Does the Investigation has a Study? (Is this actually required by spec?)
- [ ] Does the Study has an Assay? (Is this actually required by spec?)

Some fields could be checked for proper data format, including future own id/naming conventions:
- [ ] Source Name
- [ ] Sample Name
- [ ] Raw Data File
- [ ] ...
- [ ] PubMed ID, Publication DOI
- [ ] Person Email, Phone, Fax

**Comment by holtgrewe** [comment]
Yes, very important. Also Source Name and Sample Name have to be there, I believe, so we can validate files to have them and start with Source Name.

**Comment by holtgrewe** [comment]
Just had a file upload to SODAR that had an empty extract name. This should be illegal.

**Comment by mkuhring** [comment]
> Source Name and Sample Name have to be there

This is actually already done here: https://github.com/bihealth/altamisa/blob/master/altamisa/isatab/parse_assay_study.py#L632-L637
